<a href="https://colab.research.google.com/github/ynht-edu/game-sales-data/blob/main/tubes_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# download csv file from gdrive
!gdown --id 1M_3PGzqRTrmtKgd4AtaFtu2kw_xDcH15

/usr/local/lib/python3.10/dist-packages/gdown/cli.py:121: FutureWarning: Option `--id` was deprecated in version 4.3.1 and will be removed in 5.0. You don't need to pass it anymore to use a file ID.
  warnings.warn(
Downloading...
From: https://drive.google.com/uc?id=1M_3PGzqRTrmtKgd4AtaFtu2kw_xDcH15
To: /content/vgsales.csv
100% 1.36M/1.36M [00:00<00:00, 129MB/s]


In [ ]:
# install dash framework for data apps
!pip install dash

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.2/10.2 MB 16.0 MB/s eta 0:00:00


In [ ]:
# import libraries needed
import numpy as np
import pandas as pd
import plotly.express as px
import plotly.figure_factory as ff
import plotly.graph_objects as go
from dash import Dash, dcc, html, Input, Output
import torch



































In [ ]:
#
df = pd.read_csv("vgsales.csv")
df

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

In [ ]:
df.info()
df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


Rank          Year      NA_Sales      EU_Sales      JP_Sales  \
count  16598.000000  16327.000000  16598.000000  16598.000000  16598.000000   
mean    8300.605254   2006.406443      0.264667      0.146652      0.077782   
std     4791.853933      5.828981      0.816683      0.505351      0.309291   
min        1.000000   1980.000000      0.000000      0.000000      0.000000   
25%     4151.250000   2003.000000      0.000000      0.000000      0.000000   
50%     8300.500000   2007.000000      0.080000      0.020000      0.000000   
75%    12449.750000   2010.000000      0.240000      0.110000      0.040000   
max    16600.000000   2020.000000     41.490000     29.020000     10.220000   

        Other_Sales  Global_Sales  
count  16598.000000  16598.000000  
mean       0.048063      0.537441  
std        0.188588      1.555028  
min        0.000000      0.010000  
25%        0.000000      0.060000  
50%        0.010000      0.170000  
75%        0.040000      0.470000  
max       10.570000     82.740000

# Data Preprocessing

In [ ]:
# find how many data is missing for each column
df.isna().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

In [ ]:
  # find rows that contain missing value in Year column
df[df["Year"].isna() == 1]

Rank                         Name Platform  Year         Genre  \
179      180              Madden NFL 2004      PS2   NaN        Sports   
377      378             FIFA Soccer 2004      PS2   NaN        Sports   
431      432   LEGO Batman: The Videogame      Wii   NaN        Action   
470      471   wwe Smackdown vs. Raw 2006      PS2   NaN      Fighting   
607      608               Space Invaders     2600   NaN       Shooter   
...      ...                          ...      ...   ...           ...   
16307  16310                Freaky Flyers       GC   NaN        Racing   
16327  16330                    Inversion       PC   NaN       Shooter   
16366  16369  Hakuouki: Shinsengumi Kitan      PS3   NaN     Adventure   
16427  16430                 Virtua Quest       GC   NaN  Role-Playing   
16493  16496                   The Smurfs      3DS   NaN        Action   

                                    Publisher  NA_Sales  EU_Sales  JP_Sales  \
179                           Electronic Arts      4.26      0.26      0.01   
377                           Electronic Arts      0.59      2.36      0.04   
431    Warner Bros. Interactive Entertainment      1.86      1.02      0.00   
470                                       NaN      1.57      1.02      0.00   
607                                     Atari      2.36      0.14      0.00   
...                                       ...       ...       ...       ...   
16307                                 Unknown      0.01      0.00      0.00   
16327                      Namco Bandai Games      0.01      0.00      0.00   
16366                                 Unknown      0.01      0.00      0.00   
16427                                 Unknown      0.01      0.00      0.00   
16493                                 Unknown      0.00      0.01      0.00   

       Other_Sales  Global_Sales  
179           0.71          5.23  
377           0.51          3.49  
431           0.29          3.17  
470           0.41          3.00  
607           0.03          2.53  
...            ...           ...  
16307         0.00          0.01  
16327         0.00          0.01  
16366         0.00          0.01  
16427         0.00          0.01  
16493         0.00          0.01  

[271 rows x 11 columns]

In [ ]:
# find rows that contain missing value in Publisher column
df[df["Publisher"].isna() == 1]

Rank                                               Name Platform  \
470      471                         wwe Smackdown vs. Raw 2006      PS2   
1303    1305                                     Triple Play 99       PS   
1662    1664       Shrek / Shrek 2 2-in-1 Gameboy Advance Video      GBA   
2222    2224                                 Bentley's Hackpack      GBA   
3159    3161  Nicktoons Collection: Game Boy Advance Video V...      GBA   
3166    3168  SpongeBob SquarePants: Game Boy Advance Video ...      GBA   
3766    3768  SpongeBob SquarePants: Game Boy Advance Video ...      GBA   
4145    4147                                 Sonic the Hedgehog      PS3   
4526    4528  The Fairly Odd Parents: Game Boy Advance Video...      GBA   
4635    4637  The Fairly Odd Parents: Game Boy Advance Video...      GBA   
5302    5304      Dragon Ball Z: Budokai Tenkaichi 2 (JP sales)      Wii   
5647    5649  Cartoon Network Collection: Game Boy Advance V...      GBA   
6272    6274  The Legend of Zelda: The Minish Cap(weekly JP ...      GBA   
6437    6439           Sonic X: Game Boy Advance Video Volume 1      GBA   
6562    6564  Dora the Explorer: Game Boy Advance Video Volu...      GBA   
6648    6650  Cartoon Network Collection: Game Boy Advance V...      GBA   
6849    6851     All Grown Up!: Game Boy Advance Video Volume 1      GBA   
7208    7210  Nicktoons Collection: Game Boy Advance Video V...      GBA   
7351    7353                     Yu Yu Hakusho: Dark Tournament      PS2   
7470    7472  SpongeBob SquarePants: Game Boy Advance Video ...      GBA   
7953    7955                   Thomas the Tank Engine & Friends      GBA   
8330    8332    Dragon Ball GT: Game Boy Advance Video Volume 1      GBA   
8341    8343  Codename: Kids Next Door: Game Boy Advance Vid...      GBA   
8368    8370  Teenage Mutant Ninja Turtles: Game Boy Advance...      GBA   
8503    8505                                       Stronghold 3       PC   
8770    8772  Cartoon Network Collection: Game Boy Advance V...      GBA   
8848    8850  Pokémon: Johto Photo Finish: Game Boy Advance ...      GBA   
8896    8898  Strawberry Shortcake: Game Boy Advance Video V...      GBA   
9517    9519                             Farming Simulator 2011       PC   
9749    9751  Super Robot Wars OG Saga: Masou Kishin II - Re...      PSP   
10382  10384                   Disney Channel Collection Vol. 1      GBA   
10494  10496             Atsumare! Power Pro Kun no DS Koushien       DS   
11076  11078                       Action Man-Operation Extreme       PS   
11526  11528  Cartoon Network Collection: Game Boy Advance V...      GBA   
12487  12489                               Chou Soujuu Mecha MG       DS   
12517  12519       Prinny: Can I Really Be The Hero? (US sales)      PSP   
13278  13280                     Monster Hunter Frontier Online      PS3   
13672  13674                          B.L.U.E.: Legend of Water       PS   
13962  13964                                     World of Tanks     X360   
14087  14089                                       Housekeeping       DS   
14296  14299                                Bikkuriman Daijiten       DS   
14311  14314                                      Silverlicious       DS   
14698  14701                                 UK Truck Simulator       PC   
14942  14945  Umineko no Naku Koro ni San: Shinjitsu to Gens...      PS3   
15056  15059                                            Xia-Xia       DS   
15261  15264                                       Mario Tennis      3DS   
15325  15328  Nicktoons Collection: Game Boy Advance Video V...      GBA   
15353  15356                   Demolition Company: Gold Edition       PC   
15788  15791  Moshi, Kono Sekai ni Kami-sama ga Iru to suru ...      PSV   
15915  15918                                       Dream Dancer       DS   
16191  16194                    Homeworld Remastered Collection       PC   
16198  16201             AKB1/48: Idol to Guam de Koishi

# Data Analytics

In [ ]:
df[df.columns[6:11]].describe([.1, .25, .5, .75, .90])

NA_Sales      EU_Sales      JP_Sales   Other_Sales  Global_Sales
count  16598.000000  16598.000000  16598.000000  16598.000000  16598.000000
mean       0.264667      0.146652      0.077782      0.048063      0.537441
std        0.816683      0.505351      0.309291      0.188588      1.555028
min        0.000000      0.000000      0.000000      0.000000      0.010000
10%        0.000000      0.000000      0.000000      0.000000      0.020000
25%        0.000000      0.000000      0.000000      0.000000      0.060000
50%        0.080000      0.020000      0.000000      0.010000      0.170000
75%        0.240000      0.110000      0.040000      0.040000      0.470000
90%        0.610000      0.350000      0.180000      0.110000      1.210000
max       41.490000     29.020000     10.220000     10.570000     82.740000

Dari data di atas kita bisa melihat statistik kolom penjualan video game

In [ ]:
print(df["Platform"].value_counts())
print(df['Platform'].unique())
print(df['Platform'].nunique())
Platform = []
nilai_unik_frekuensi = df['Platform'].value_counts()
for nilai, frekuensi in nilai_unik_frekuensi.items():
    Platform.append(f"{nilai} ({frekuensi})")
Platform_string = ', '.join(Platform)
print(Platform_string)

DS      2163
PS2     2161
PS3     1329
Wii     1325
X360    1265
PSP     1213
PS      1196
PC       960
XB       824
GBA      822
GC       556
3DS      509
PSV      413
PS4      336
N64      319
SNES     239
XOne     213
SAT      173
WiiU     143
2600     133
NES       98
GB        98
DC        52
GEN       27
NG        12
SCD        6
WS         6
3DO        3
TG16       2
GG         1
PCFX       1
Name: Platform, dtype: int64
['Wii' 'NES' 'GB' 'DS' 'X360' 'PS3' 'PS2' 'SNES' 'GBA' '3DS' 'PS4' 'N64'
 'PS' 'XB' 'PC' '2600' 'PSP' 'XOne' 'GC' 'WiiU' 'GEN' 'DC' 'PSV' 'SAT'
 'SCD' 'WS' 'NG' 'TG16' '3DO' 'GG' 'PCFX']
31
DS (2163), PS2 (2161), PS3 (1329), Wii (1325), X360 (1265), PSP (1213), PS (1196), PC (960), XB (824), GBA (822), GC (556), 3DS (509), PSV (413), PS4 (336), N64 (319), SNES (239), XOne (213), SAT (173), WiiU (143), 2600 (133), NES (98), GB (98), DC (52), GEN (27), NG (12), SCD (6), WS (6), 3DO (3), TG16 (2), GG (1), PCFX (1)


In [ ]:
print(df["Year"].isnull().mean() * 100)
print(str(df['Year'].min()) + ' - ' + str(df['Year'].max()))

1.6327268345583803
1980.0 - 2020.0


In [ ]:
print(df["Genre"].value_counts())
print(df['Genre'].unique())
Platform = []
nilai_unik_frekuensi = df['Genre'].value_counts()
for nilai, frekuensi in nilai_unik_frekuensi.items():
    Platform.append(f"{nilai} ({frekuensi})")
Platform_string = ', '.join(Platform)
print(Platform_string)

Action          3316
Sports          2346
Misc            1739
Role-Playing    1488
Shooter         1310
Adventure       1286
Racing          1249
Platform         886
Simulation       867
Fighting         848
Strategy         681
Puzzle           582
Name: Genre, dtype: int64
['Sports' 'Platform' 'Racing' 'Role-Playing' 'Puzzle' 'Misc' 'Shooter'
 'Simulation' 'Action' 'Fighting' 'Adventure' 'Strategy']
Action (3316), Sports (2346), Misc (1739), Role-Playing (1488), Shooter (1310), Adventure (1286), Racing (1249), Platform (886), Simulation (867), Fighting (848), Strategy (681), Puzzle (582)


In [ ]:
print(df["Publisher"].value_counts())
print(df['Publisher'].unique())
print(df['Publisher'].nunique())
array = []
nilai_unik_frekuensi = df['Publisher'].value_counts()
for nilai, frekuensi in nilai_unik_frekuensi.items():
    array.append(f"{nilai} ({frekuensi})")
print(array)

Electronic Arts                 1351
Activision                       975
Namco Bandai Games               932
Ubisoft                          921
Konami Digital Entertainment     832
                                ... 
Warp                               1
New                                1
Elite                              1
Evolution Games                    1
UIG Entertainment                  1
Name: Publisher, Length: 578, dtype: int64
['Nintendo' 'Microsoft Game Studios' 'Take-Two Interactive'
 'Sony Computer Entertainment' 'Activision' 'Ubisoft' 'Bethesda Softworks'
 'Electronic Arts' 'Sega' 'SquareSoft' 'Atari' '505 Games' 'Capcom'
 'GT Interactive' 'Konami Digital Entertainment'
 'Sony Computer Entertainment Europe' 'Square Enix' 'LucasArts'
 'Virgin Interactive' 'Warner Bros. Interactive Entertainment'
 'Universal Interactive' 'Eidos Interactive' 'RedOctane' 'Vivendi Games'
 'Enix Corporation' 'Namco Bandai Games' 'Palcom' 'Hasbro Interactive'
 'THQ' 'Fox Interactive' 'A

Dari data di atas kita bisa melihat publisher video game yang ada pada file vgsales.csv

In [ ]:
print(df["NA_Sales"].isnull().mean() * 100)
print(str(df['NA_Sales'].min()) + ' - ' + str(df['NA_Sales'].max()))

0.0
0.0 - 41.49


In [ ]:
print(df["Year"].value_counts())
Platform = []
nilai_unik_frekuensi = df['Year'].value_counts()
for nilai, frekuensi in nilai_unik_frekuensi.items():
    Platform.append(f"{nilai} ({frekuensi})")
Platform_string = ', '.join(Platform)
print(Platform_string)

2009.0    1431
2008.0    1428
2010.0    1259
2007.0    1202
2011.0    1139
2006.0    1008
2005.0     941
2002.0     829
2003.0     775
2004.0     763
2012.0     657
2015.0     614
2014.0     582
2013.0     546
2001.0     482
1998.0     379
2000.0     349
2016.0     344
1999.0     338
1997.0     289
1996.0     263
1995.0     219
1994.0     121
1993.0      60
1981.0      46
1992.0      43
1991.0      41
1982.0      36
1986.0      21
1989.0      17
1983.0      17
1990.0      16
1987.0      16
1988.0      15
1985.0      14
1984.0      14
1980.0       9
2017.0       3
2020.0       1
Name: Year, dtype: int64
2009.0 (1431), 2008.0 (1428), 2010.0 (1259), 2007.0 (1202), 2011.0 (1139), 2006.0 (1008), 2005.0 (941), 2002.0 (829), 2003.0 (775), 2004.0 (763), 2012.0 (657), 2015.0 (614), 2014.0 (582), 2013.0 (546), 2001.0 (482), 1998.0 (379), 2000.0 (349), 2016.0 (344), 1999.0 (338), 1997.0 (289), 1996.0 (263), 1995.0 (219), 1994.0 (121), 1993.0 (60), 1981.0 (46), 1992.0 (43), 1991.0 (41), 1982.0 (36

In [ ]:
df.sort_values(by=["Year", "Global_Sales"], ascending = False)[0:10]

Rank                                              Name Platform  \
5957    5959                            Imagine: Makeup Artist       DS   
14390  14393  Phantasy Star Online 2 Episode 4: Deluxe Package      PS4   
16241  16244  Phantasy Star Online 2 Episode 4: Deluxe Package      PSV   
16438  16441                  Brothers Conflict: Precious Baby      PSV   
221      222                                           FIFA 17      PS4   
271      272                        Uncharted 4: A Thief's End      PS4   
351      352                         Tom Clancy's The Division      PS4   
770      772                                   Far Cry: Primal      PS4   
845      847                         Tom Clancy's The Division     XOne   
1026    1028                                         Overwatch      PS4   

         Year         Genre                    Publisher  NA_Sales  EU_Sales  \
5957   2020.0    Simulation                      Ubisoft      0.27      0.00   
14390  2017.0  Role-Playing                         Sega      0.00      0.00   
16241  2017.0  Role-Playing                         Sega      0.00      0.00   
16438  2017.0        Action                 Idea Factory      0.00      0.00   
221    2016.0        Sports              Electronic Arts      0.28      3.75   
271    2016.0       Shooter  Sony Computer Entertainment      1.30      2.07   
351    2016.0       Shooter                      Ubisoft      1.28      1.61   
770    2016.0        Action                      Ubisoft      0.59      1.16   
845    2016.0       Shooter                      Ubisoft      1.20      0.62   
1026   2016.0       Shooter                   Activision      0.64      0.68   

       JP_Sales  Other_Sales  Global_Sales  
5957       0.00         0.02          0.29  
14390      0.03         0.00          0.03  
16241      0.01         0.00          0.01  
16438      0.01         0.00          0.01  
221        0.06         0.69          4.77  
271        0.18         0.65          4.20  
351        0.15         0.57          3.61  
770        0.06         0.33          2.13  
845        0.00         0.18          2.01  
1026       0.14         0.26          1.73

In [ ]:
df.sort_values(by=["Year", "Global_Sales"], ascending = [1, 0])[0:10]

Rank             Name Platform    Year     Genre   Publisher  NA_Sales  \
258    259        Asteroids     2600  1980.0   Shooter       Atari      4.00   
544    545  Missile Command     2600  1980.0   Shooter       Atari      2.56   
1766  1768          Kaboom!     2600  1980.0      Misc  Activision      1.07   
1969  1971         Defender     2600  1980.0      Misc       Atari      0.99   
2669  2671           Boxing     2600  1980.0  Fighting  Activision      0.72   
4025  4027       Ice Hockey     2600  1980.0    Sports  Activision      0.46   
5366  5368          Freeway     2600  1980.0    Action  Activision      0.32   
6317  6319           Bridge     2600  1980.0      Misc  Activision      0.25   
6896  6898         Checkers     2600  1980.0      Misc       Atari      0.22   
239    240         Pitfall!     2600  1981.0  Platform  Activision      4.21   

      EU_Sales  JP_Sales  Other_Sales  Global_Sales  
258       0.26       0.0         0.05          4.31  
544       0.17       0.0         0.03          2.76  
1766      0.07       0.0         0.01          1.15  
1969      0.05       0.0         0.01          1.05  
2669      0.04       0.0         0.01          0.77  
4025      0.03       0.0         0.01          0.49  
5366      0.02       0.0         0.00          0.34  
6317      0.02       0.0         0.00          0.27  
6896      0.01       0.0         0.00          0.24  
239       0.24       0.0         0.05          4.50

In [ ]:
df.sort_values(by=["Global_Sales", "NA_Sales"], ascending = False)[0:10]

Rank                       Name Platform    Year         Genre Publisher  \
0     1                 Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2          Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3             Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4          Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5   Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   
5     6                     Tetris       GB  1989.0        Puzzle  Nintendo   
6     7      New Super Mario Bros.       DS  2006.0      Platform  Nintendo   
7     8                   Wii Play      Wii  2006.0          Misc  Nintendo   
8     9  New Super Mario Bros. Wii      Wii  2009.0      Platform  Nintendo   
9    10                  Duck Hunt      NES  1984.0       Shooter  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37  
5     23.20      2.26      4.22         0.58         30.26  
6     11.38      9.23      6.50         2.90         30.01  
7     14.03      9.20      2.93         2.85         29.02  
8     14.59      7.06      4.70         2.26         28.62  
9     26.93      0.63      0.28         0.47         28.31

In [ ]:
df.groupby("Publisher")["Publisher"].count().reset_index(name="Count").sort_values(by="Count", ascending = True)

Publisher  Count
287                         Locus      1
349                  Navarre Corp      1
344               NDA Productions      1
341                      Mystique      1
141                         Elite      1
..                            ...    ...
275  Konami Digital Entertainment    823
524                       Ubisoft    918
347            Namco Bandai Games    928
21                     Activision    966
138               Electronic Arts   1339

[576 rows x 2 columns]

In [ ]:
df.groupby("Platform")["Platform"].value_count().reset_index(name="Count").sort_values(by="Count", ascending = True)

Platform  Count
9        GG      1
14     PCFX      1
24     TG16      2
1       3DO      3
25       WS      6
22      SCD      6
12       NG     12
8       GEN     27
3        DC     52
5        GB     97
11      NES     98
0      2600    116
27     WiiU    143
21      SAT    173
30     XOne    213
23     SNES    239
10      N64    316
18      PS4    336
20      PSV    410
2       3DS    499
7        GC    542
6       GBA    786
29       XB    803
13       PC    938
15       PS   1189
19      PSP   1197
28     X360   1234
26      Wii   1290
17      PS3   1304
16      PS2   2127
4        DS   2131

# Data Visualizations

In [ ]:
fig = px.treemap(df, path=["Platform", "Genre"], title="Grafik Treemap Hierarki Platform dan Genre")
fig.show()

Dari sini bisa kita dapat melihat distribusi genre video games di setiap platformnya. Dapat dilihat bahwa game dengan genre action dan sports mendominasi di banyak platform.



In [ ]:
# exclude top 10 platform
value_counts = df['Platform'].value_counts()
to_remove = value_counts[value_counts >= 600].index
df2 = df[~df.Platform.isin(to_remove)]

fig = px.treemap(df2, path=["Platform", "Genre"], title="Grafik Treemap Hierarki Platform dan Genre (di luar platform top 10)")
fig.show()

Dari grafik treemap di atas, kita dapat melihat lebih jelas distribusi genre video games berdasarkan platformnya untuk platform di bawah top 10 dengan judul game yang dirilis terbanyak.


In [ ]:
df1 = df
df1['count'] = df.groupby("Genre")["Platform"].transform(len)
fig = px.parallel_categories(df1, dimensions=["Platform", "Genre"], color = "count",
                             color_continuous_scale=px.colors.sequential.Inferno,
                             title="Diagram Kategori Paralel")
fig.show()

Grafik di atas menunjukkan hubungan sekaligus perbandingan data. Garis yang ditampilkan menunjukkan hubungan data sedangkan warna dan ukuran kotak menunjukkan kuantitas dari data yang terhubung. Dapat diketahui bahwa game dengan genre action memiliki judul game yang dirilis terbanyak dan hubungannya dengan platform yang merilis game tersebut

In [ ]:
# visualize the distribution of games released
fig = px.histogram(df, x = "Year", title = "Grafik Persebaran Jumlah Game yang Dirilis Setiap Tahun")
fig.show()

Dari grafik persebaran di atas, dapat kita ketahui bahwa jumlah game yang dirilis tertinggi terjadi pada tahun 2008 sampai 2010 dan jumlah game yang dirilis terendah terjadi pada tahun 1983 sampai 1990

In [ ]:
fig = px.line(df.groupby(["Year"])["Global_Sales"].sum().reset_index(name="Sales_Sum"), x = "Year", y =["Sales_Sum"],
              title="Grafik Penjualan Game Global Setiap Tahun", labels={"Year":"Tahun", "value":"Jumlah Penjualan Game GLobal (dalam juta)",
              "Sales_Sum":"Jumlah Penjualan Game Global"})
fig.show()

In [ ]:
df_sum = df.groupby(["Year"]).sum().reset_index()
fig = px.area(title="Grafik Penjualan Game Global Setiap Tahun",
              labels={"Year":"Tahun", "value":"Jumlah Penjualan Game GLobal (dalam juta)", "Sales_Sum":"Jumlah Penjualan Game Global"})

fig.add_trace(go.Scatter(
    name = 'NA Sales',
    x = df_sum["Year"],
    y = df_sum["NA_Sales"],
    stackgroup='one'
   ))
fig.add_trace(go.Scatter(
    name = 'EU Sales',
    x = df_sum["Year"],
    y = df_sum["EU_Sales"],
    stackgroup='one'
   ))
fig.add_trace(go.Scatter(
    name = 'JP Sales',
    x = df_sum["Year"],
    y = df_sum["JP_Sales"],
    stackgroup='one'
   ))
fig.add_trace(go.Scatter(
    name = 'Other Sales',
    x = df_sum["Year"],
    y = df_sum["Other_Sales"],
    stackgroup='one'
   ))
fig.show()


<ipython-input-26-d0ad41045226>:1: FutureWarning:

The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.



Dari grafik area di atas, dapat kita lihat bahwa distribusi penjualan game setiap tahunnya di masing-masing area mencapai tingkat tertingginya di tahun 2005 sampai dengan tahun 2010.

In [ ]:
df1 = df.groupby("Platform")["Global_Sales"].sum().reset_index(name="Sales_Sum").sort_values(by="Sales_Sum", ascending = True)
fig = px.histogram(df1, x = "Sales_Sum", y = "Platform", orientation="h", title="Jumlah Penjualan Game Global Berdasarkan Platform", labels={"Sales_Sum":"Penjualan (dalam jutaan)"})
fig.show()

In [ ]:
# visualize ratio of games based on the platform
fig = px.pie(df.groupby("Platform")["Platform"].count().reset_index(name="count"), values = "count", names = "Platform", title = "Pie Chart Jumlah Game yang Dirilis Pada Setiap Platform")
fig.show()

Dari pie chart di atas, dapat diketahui dengan jelas perbandingan jumlah game yang dirilis pada setiap platform dengan persentase yang ditampilkan. Terlihat bahwa platform dengan jumlah judul game yang dirilis terbanyak adalah DS disusul oleh PS2.

In [ ]:
# visualize ratio and total games released from every publishers
df1 = df.groupby("Publisher")["Publisher"].count().reset_index(name="count")
df1 = df1.loc[df1["count"] > .02*df["Publisher"].count()]
df2 = df.groupby("Publisher")["Publisher"].count().reset_index(name="count")
df2 = df2.loc[df2["count"] <= .02*df["Publisher"].count()]
df2 = pd.DataFrame({"Publisher":["Other"], "count":[df2["count"].sum().sum()]})
TGByPublisher = pd.concat([df1, df2], axis=0)

fig = px.pie(TGByPublisher, values="count", names="Publisher", title="Perbandingan Jumlah Game yang Dirilis Berdasarkan Publisher")
fig.show()

Dari pie chart di atas, dapat diketahui dengan jelas perbandingan jumlah game yang dirilis oleh setiap publisher dengan persentase yang ditampilkan. Terlihat bahwa publisher dengan jumlah judul game yang dirilis terbanyak adalah Electronic Arts (EA). Dengan mempertimbangkan persentase yang tertampil, dapat diketahui bahwa jumlah judul game yang dirilis oleh publisher lebih menyebar merata.

In [ ]:
fig = px.bar(df.groupby("Genre")["Genre"].count().reset_index(name="count"), x = "Genre", y = "count",
             title="Grafik Jumlah Game yang Rilis Berdasarkan Genre", labels={"count":"Jumlah Game"})
fig.show()

Dari grafik bar chart dia atas dapat kita ketahui bahwa game dengan genre action adalah game dengan jumlah game yang dirilis terbanyak sedangkan game dengan genre puzzle adalah game dengan jumlah game yang dirilis paling sedikit.

In [ ]:
fig = px.bar(df.groupby("Genre")["Global_Sales"].sum().reset_index(name="Sales"), x = "Genre", y = "Sales",
             title="Grafik Penjualan Game Global Berdasarkan Genre", labels={"Sales":"Penjualan (dalam jutaan)"})
fig.show()

Pada grafik bar chart penjualan, dapat kita ketahui bahwa game dengan genre action adalah game dengan jumlah penjualan global terbanyak sedangkan game dengan genre strategy adalah game dengan jumlah penjualan global paling sedikit. Bisa dilihat bahwa terdapat beberapa perbedaan dibandingkan dengan grafik bar chart jumlah game yang dirilis.

In [ ]:
df_sum = df.groupby(["Genre"]).sum().reset_index()
fig = go.Figure(
    data=[
        go.Bar(
          name = 'NA Sales',
          x = df_sum["Genre"],
          y = df_sum["NA_Sales"],
          offsetgroup='one'
        ),
        go.Bar(
          name = 'EU Sales',
          x = df_sum["Genre"],
          y = df_sum["EU_Sales"],
          offsetgroup='one',
          base = df_sum["NA_Sales"]
        ),
        go.Bar(
          name = 'JP Sales',
          x = df_sum["Genre"],
          y = df_sum["JP_Sales"],
          offsetgroup='one',
          base = df_sum["NA_Sales"] + df_sum["EU_Sales"]
        ),
        go.Bar(
          name = 'Other Sales',
          x = df_sum["Genre"],
          y = df_sum["Other_Sales"],
          offsetgroup='one',
          base = df_sum["NA_Sales"] + df_sum["EU_Sales"] + df_sum["JP_Sales"]
        )
    ],
    layout=go.Layout(
        title="Grafik Penjualan Game Global Berdasarkan Genre",
        yaxis_title="Penjualan (dalam jutaan)"
    )
)

fig.show()

<ipython-input-32-e02075ea72e1>:1: FutureWarning:

The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.



Dari grafik bar chart dia atas dapat kita ketahui bahwa game dengan genre action adalah game dengan jumlah penjualan global terbanyak sedangkan game dengan genre strategy adalah game dengan jumlah penjualan global paling sedikit. Bisa dilihat bahwa terdapat beberapa perbedaan dibandingkan dengan grafik bar chart jumlah game yang dirilis.


In [ ]:
df1 = df.groupby(["Genre", "Year"])["Genre"].count().reset_index(name="Count")

fig = px.bar(df1, x = "Genre", y = "Count",
             animation_frame="Year", animation_group="Genre",
             range_x=[-1, 12], range_y=[0,300],
             title="Grafik Penjualan Game Global Berdasarkan Genre", labels={"Count":"Jumlah Game Yang Dirilis"})
fig.show()

In [ ]:
df1 = df.groupby(["Genre", "Year"])["Global_Sales"].sum().reset_index(name="Sales")

fig = px.bar(df1, x = "Genre", y = "Sales",
             animation_frame="Year", animation_group="Genre",
             range_x=[-1, 12], range_y=[0,150],
             title="Grafik Penjualan Game Global Berdasarkan Genre", labels={"Sales":"Penjualan (dalam jutaan)"})
fig.show()

In [ ]:
GSByPublisher = df.groupby(["Publisher"])["Global_Sales"].sum().reset_index(name="Global_Sales").sort_values(by="Global_Sales", ascending = False)
fig = px.bar(GSByPublisher, x = "Publisher", y = "Global_Sales", labels={"Global_Sales":"Jumlah Penjualan Global (dalam juta)"},
             title="Grafik Penjualan Game Global Berdasarkan Publisher")
fig.show()

Dari grafik pie chart di atas, secara spesifik, dapat kita ketahui bahwa electronic arts merupakan publisher yang merilis game paling banyak, namun jika kita bandingkan dengan semua persentase. Publisher lain-lain diluar data ini memiliki persentase terbesar sebesar 46% dan publisher dengan perilisan game paling sedikit adalah Tecmo Koei. Pada grafik bar chart, kita dapat melihat urutan publisher dengan penjualan game dari yang terbesar hingga terkecil dengan Nintendo sebagai publisher dengan penjualan terbesar sedangkan Commseed sebagai publisher dengan penjualan terkecil

In [ ]:
df4 = GSByPublisher[GSByPublisher['Global_Sales'] > 50]
fig = px.bar(df4, x = "Publisher", y = "Global_Sales", labels={"Global_Sales":"Jumlah Penjualan Global (dalam juta)"},
             title="Grafik Penjualan Game Global Di Atas 50 Juta Copy Berdasarkan Publisher")
fig.show()

Grafik diatas adalah plot distribusi penjualan global secara keseluruhan. Dapat diketahui bahwa distribusi penjualan game secara keseluruhan (tidak dipisahkan dengan genre atau publisher) padat di angka penjualan 0—10 juta copy.

In [ ]:
df4 = GSByPublisher[GSByPublisher['Global_Sales'] < 50]
df4 = df4[df4['Global_Sales'] > 1]
fig = px.bar(df4, x = "Publisher", y = "Global_Sales", labels={"Global_Sales":"Jumlah Penjualan Global (dalam jutaan)"},
             title="Grafik Penjualan Game Global Di Bawah 50 Juta dan Di Atas 1 Juta Copy Berdasarkan Publisher")
fig.show()

Data di atas merupakan distribusi penjualan game di atas satu juta copy dan di bawah 50 juta copy yang dibedakan berdasarkan publisher-nya. Data di atas diurutkan secara descending atau dimulai dari data yang terbesar hingga yang terkecil. Dari data di atas, Codemasters menjadi publisher dengan penjualan terbanyak di atas satu juta copy dan di bawah 50 juta copy (sekitar 47,87 juta penjualan).

In [ ]:
df4 = GSByPublisher[GSByPublisher['Global_Sales'] < 1]
fig = px.bar(df4, x = "Publisher", y = "Global_Sales", labels={"Global_Sales":"Jumlah Penjualan Global (dalam juta)"},
             title="Grafik Penjualan Game Global Di Bawah 1 Juta Copy Berdasarkan Publisher")
fig.show()

Data di atas merupakan distribusi penjualan game di bawah satu juta copy yang dibedakan berdasarkan publisher-nya. Data di atas diurutkan secara descending atau dimulai dari data yang terbesar hingga yang terkecil. Dari data di atas, SNK Playmore menjadi publisher dengan penjualan terbanyak di bawah satu juta (sekitar 960 ribu penjualan).

In [ ]:
x = [[1, 5, 7, 8, 3]]
fig = ff.create_distplot(hist_data = [df.Global_Sales.values.tolist()], group_labels=["Distribusi Penjualan Global"],
                         bin_size = [.5])
fig.update_layout(title="Grafik Distribusi Penjualan Global")
fig.show()

Grafik diatas adalah plot distribusi penjualan global secara keseluruhan. Dapat diketahui bahwa distribusi penjualan game secara keseluruhan (tidak dipisahkan dengan genre atau publisher) padat di angka penjualan 0—10 juta copy.

In [ ]:
x_data = ["Penjualan Amerika Utara", "Penjualan Uni-Eropa", "Penjualan Jepang", "Penjualan Lainnya", "Penjualan Global",]

y_data = []
for data in df.columns[6:11]:
  y_data.append(df[data])

fig = go.Figure()
for xd, yd in zip(x_data, y_data):
  fig.add_trace(go.Box(y = yd, name = xd, boxpoints = "all", jitter=0.5, whiskerwidth=0.2, marker_size=2, line_width=1)  )

fig.update_layout(title='Grafik Distribusi Penjualan Video Game (Box and Whisker)',
    yaxis=dict(autorange=True, showgrid=True, zeroline=True, dtick=5, gridcolor='rgb(255, 255, 255)', gridwidth=1, zerolinecolor='rgb(255, 255, 255)', zerolinewidth=2)
)

#fig = px.box(df, y=["Global_Sales"], labels={"Global_Sales":"Penjualan Global"}, title="Grafik Distribusi Penjualan Game Global (Box and Whisker)")
fig.show()

Dari grafik box and whiskers di atas dapat kita lihat bahwa distribusi penjualan global sangat timpang dan didominasi dengan penjualan game global yang memiliki nilai 0 sampai 5 juta *copy*. Grafik diatas adalah grafik interaktif sehingga persebaran data dapat diketahui dengan meng-hover bagian box pada grafik.

In [ ]:
Count_Corr = df.groupby(["Publisher"])["Global_Sales"].sum().reset_index(name="Sales")
Sales_Corr = df.groupby(["Publisher"])["Global_Sales"].count().reset_index(name="Count")
Temp = pd.concat([Count_Corr, Sales_Corr["Count"]], axis=1)
fig = px.scatter(Temp, x = "Count", y="Sales", color="Publisher", trendline="ols", trendline_scope="overall",
                 labels={"Sales":"Jumlah Penjualan Game Yang Dirilis (dalam juta)", "Count":"Jumlah Judul Game Yang Dirilis"})
fig.update_layout(title="Grafik Scatter Jumlah Dan Penjualan Game Yang Dirilis Berdasarkan Publisher")
fig.show()

Dari grafik scatter jumlah dan penjualan game yang dirilis berdasarkan publisher di atas, dapat diketahui bahwa jumlah judul game cenderung berbanding lurus dengan jumlah penjuala game yang dirilis dengan nilai koefisien determinasi (R^2) bernilai 0,639834.

In [ ]:
Count_Corr = df.groupby(["Genre"])["Global_Sales"].sum().reset_index(name="Sales")
Sales_Corr = df.groupby(["Genre"])["Global_Sales"].count().reset_index(name="Count")
Temp = pd.concat([Count_Corr, Sales_Corr["Count"]], axis=1)
fig = px.scatter(Temp, x = "Count", y="Sales", color="Genre", trendline="ols", trendline_scope="overall",
                 labels={"Sales":"Jumlah Penjualan Game Yang Dirilis (dalam juta)", "Count":"Jumlah Judul Game Yang Dirilis"})
fig.update_layout(title="Grafik Scatter Jumlah Dan Penjualan Game Yang Dirilis Berdasarkan Genre")
fig.show()

Dari grafik scatter jumlah dan penjualan game yang dirilis berdasarkan genre di atas, dapat diketahui lebih jelas bahwa jumlah judul game cenderung berbanding lurus dengan jumlah penjuala game yang dirilis dengan nilai koefisien determinasi (R^2) bernilai 0,790709.

In [ ]:
Count_Corr = df.groupby(["Publisher", "Year"])["Global_Sales"].sum().reset_index(name="Sales")
Sales_Corr = df.groupby(["Publisher", "Year"])["Global_Sales"].count().reset_index(name="Count")
Temp = pd.concat([Count_Corr, Sales_Corr["Count"]], axis=1)
fig = px.scatter(Temp, x = "Count", y="Sales", color="Publisher",
                 animation_frame="Year", animation_group="Publisher",
                 range_x=[0,250],range_y=[0,250],
                 labels={"Sales":"Jumlah Penjualan Game Yang Dirilis (dalam juta)", "Count":"Jumlah Judul Game Yang Dirilis"})
fig.update_layout(title="Grafik Scatter Jumlah Dan Penjualan Game Yang Dirilis (setiap tahun)")
fig.show()

In [ ]:
Count_Corr = df.groupby(["Genre", "Year"])["Global_Sales"].sum().reset_index(name="Sales")
Sales_Corr = df.groupby(["Genre", "Year"])["Global_Sales"].count().reset_index(name="Count")
Temp = pd.concat([Count_Corr, Sales_Corr["Count"]], axis=1)
fig = px.scatter(Temp, x = "Count", y="Sales", color="Genre",
                 animation_frame="Year", animation_group="Genre",
                 range_x=[0, 300], range_y=[0,150],
                 labels={"Sales":"Jumlah Penjualan Game Yang Dirilis (dalam juta)", "Count":"Jumlah Judul Game Yang Dirilis"})
fig.update_layout(title="Grafik Scatter Jumlah Dan Penjualan Game Yang Dirilis Berdasarkan Genre (setiap tahun)")
fig.show()

In [ ]:
# show data correlations for every column
df1 = df[df.columns[0:11]]
df1.corr()

<ipython-input-45-f119f9cb5086>:3: FutureWarning:

The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.



Rank      Year  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
Rank          1.000000  0.178814 -0.401362 -0.379123 -0.267785    -0.332986   
Year          0.178814  1.000000 -0.091402  0.006014 -0.169316     0.041058   
NA_Sales     -0.401362 -0.091402  1.000000  0.767727  0.449787     0.634737   
EU_Sales     -0.379123  0.006014  0.767727  1.000000  0.435584     0.726385   
JP_Sales     -0.267785 -0.169316  0.449787  0.435584  1.000000     0.290186   
Other_Sales  -0.332986  0.041058  0.634737  0.726385  0.290186     1.000000   
Global_Sales -0.427407 -0.074735  0.941047  0.902836  0.611816     0.748331   

              Global_Sales  
Rank             -0.427407  
Year             -0.074735  
NA_Sales          0.941047  
EU_Sales          0.902836  
JP_Sales          0.611816  
Other_Sales       0.748331  
Global_Sales      1.000000

In [ ]:
fig = px.imshow(df[df.columns[0:11]].corr(), text_auto = True, title="Grafik Kekorelasian Antarkolom (heatmap)")
fig.show()

<ipython-input-46-ab6b90d975cc>:1: FutureWarning:

The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.



Dengan menggunakan heatmap, kita dapat melihat kekorelasian antarkolom yang dibantu oleh penggunaan warna dan angka kekorelasian. Dapat kita simpulkan bahwa data penjualan berkorelasi dengan penjualan lainnya. Sedangkan peringkat dan tahun cenderung tidak berkorelasi dengan data lainnya